src/main/java/seedu/address/model/module/grade/GradeTracker.java

```diff
@@ -2,24 +2,40 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
 /**
  * Represents an association class that tracks the assignments and grade for a module.
  */
 public class GradeTracker {
     public static final String MESSAGE_INVALID_GRADE =
             "Grades should be provided in the range from 0.00 to 1.00.";
+    public static final String MESSAGE_INVALID_GRADEPOINT =
+            "GradePoint should be given as a decimal from 0.00 to 5.00.";
     public static final String MESSAGE_DUPLICATE_ASSIGNMENT =
             "Assignments cannot be repeated.";
+
     private final List<Assignment> assignments;
     private Grade grade;
+    private Optional<GradePoint> gradePoint;
 
```

Comment:
Just for additional information, I read that it is not recommended to use `Optional` as the type of the property of an `Object`. There are several explanations about this which I found from :

https://stackoverflow.com/questions/56457245/why-isnt-optional-used-for-instance-variables#:~:text=Optional%20is%20a%20valuable%20specification.&text=One%20reason%20not%20to%20use,reference%20itself%20to%20be%20null%20.

https://stackoverflow.com/questions/29033518/is-it-a-good-practice-to-use-optional-as-an-attribute-in-a-class

In addition, the official Oracle Java Documentation also states the following :

"Optional is primarily intended for use as a method return type where there is a clear need to represent "no result," and where using null is likely to cause errors. A variable whose type is Optional should never itself be null; it should always point to an Optional instance."

Link : https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html#map(java.util.function.Function